v2/apiserver/internal/core/jobs.go

```diff
@@ -253,7 +300,29 @@ func (j *jobsService) Create(
 	if err != nil {
 		return errors.Wrapf(err, "error retrieving event %q from store", eventID)
 	}
-	if _, ok := event.Worker.Job(job.Name); ok {
+	if originalJob, ok := event.Worker.Job(job.Name); ok {
+		// If we have a match and this is a retry, the job must be one pre-selected
+		// as eligible for caching (e.g., it has succeeded and does not use a
+		// shared workspace).
+		if event.Labels != nil || event.Labels[RetryLabelKey] != "" {
+			// However, we want to compare this job configuration to the original.
+			// If it differs, we'll want to return an error
+			if !originalJob.DeepEquals(job) {
+				return &meta.ErrConflict{
+					Type: JobKind,
+					ID:   job.Name,
+					Reason: fmt.Sprintf(
+						"Job %q is not eligible for caching, as it is not equivalent "+
+							"to the original job from event %q",
+						job.Name,
+						event.Labels[RetryLabelKey],
+					),
+				}
+			}
```

Comment:
I'm ok with this behavior, at least near term, but I thought you had wanted to throw away the inherited result in this scenario and proceed with scheduling a new job.

Reply:
Oh yup, had forgotten.  Logic updated.